src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -0,0 +1,189 @@
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { StoreCreditActionCreator } from '../../../store-credit';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { StripeElement, StripeElements, StripeElementsOptions, StripeUPEClient } from './stripe-upe';
+import StripeUPEPaymentInitializeOptions from './stripe-upe-initialize-options';
+import StripeUPEScriptLoader from './stripe-upe-script-loader';
+
+export default class StripeUPEPaymentStrategy implements PaymentStrategy {
+    private _initializeOptions?: StripeUPEPaymentInitializeOptions;
+    private _stripeUPEClient?: StripeUPEClient;
+    private _stripeElements?: StripeElements;
+    private _stripeElement?: StripeElement;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _stripeScriptLoader: StripeUPEScriptLoader,
+        private _storeCreditActionCreator: StoreCreditActionCreator,
+        private _locale: string
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { stripeupe, methodId, gatewayId } = options;
+
+        if (!gatewayId) {
+            throw new InvalidArgumentError('Unable to initialize payment because "gatewayId" argument is not provided.');
+        }
+
+        this._initializeOptions = stripeupe;
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(`${gatewayId}?method=${methodId}`));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { initializationData: { stripePublishableKey, stripeConnectedAccount } } = paymentMethod;
+
+        this._stripeUPEClient = await this._loadStripeJs(stripePublishableKey, stripeConnectedAccount);
+        this._stripeElement = await this._mountCardFields(paymentMethod.clientToken);
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+
+        if (!payment || !payment.paymentData) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        const { paymentData, methodId } = payment;
+        const { shouldSaveInstrument, shouldSetAsDefaultInstrument } = isHostedInstrumentLike(paymentData) ? paymentData : { shouldSaveInstrument: false, shouldSetAsDefaultInstrument: false };
```

Comment:
it is not exactly the same meaning, but would this be somewhat equivalent?

```suggestion
        const { shouldSaveInstrument = false, shouldSetAsDefaultInstrument = false } = isHostedInstrumentLike(paymentData) ? paymentData : {};
```